streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -584,6 +579,7 @@ public boolean test(final K1 key, final V1 value) {
         String name = topology.newName(LEFTJOIN_NAME);
 
         topology.addProcessor(name, new KStreamKTableLeftJoin<>((KTableImpl<K, ?, V1>) other, joiner), this.name);
+        topology.connectProcessorAndStateStores(name, other.getStoreName());
```

Comment:
Why a `null` check would fix the issue?? What I meant is that, for some cases like I mentioned above the processor node would actually try to access other state stores other than the `other.getStoreName()` and hence a `TopologyException` will be thrown.